runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,420 @@
+import net.runelite.api.Varbits;
+import net.runelite.api.events.CommandExecuted;
+import net.runelite.api.events.ConfigChanged;
+import static net.runelite.client.callback.Hooks.log;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import org.apache.commons.lang3.StringUtils;
+import org.kitteh.irc.client.library.Client;
+import org.kitteh.irc.client.library.event.channel.ChannelMessageEvent;
+import org.kitteh.irc.client.library.feature.twitch.TwitchDelaySender;
+import org.kitteh.irc.client.library.feature.twitch.TwitchListener;
+import org.kitteh.irc.client.library.feature.twitch.event.RoomStateEvent;
+import org.kitteh.irc.client.library.feature.twitch.event.UserNoticeEvent;
+import org.kitteh.irc.client.library.feature.twitch.event.UserStateEvent;
+
+@PluginDescriptor(
+		name = "Twitch",
+		enabledByDefault = false
+)
+public class TwitchPlugin extends Plugin
+{
+	@Inject
+	private net.runelite.api.Client client;
+
+	@Inject
+	private TwitchConfig config;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	private Client irc;
+
+	private boolean subscribed;
+
+	private boolean mod;
+
+	private boolean broadcaster;
+
+	private boolean subsOnly;
+
+	private boolean emoteOnly;
+
+	private String badges = "";
+
+	@Provides
+	TwitchConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(TwitchConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		twitchShutdown(irc);
+		irc = twitchSetup(config.oauthToken(), config.channelName(), config.nickName());
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		twitchShutdown(irc);
+	}
+
+	@Subscribe
+	public void onConfigChanged(ConfigChanged event)
+	{
+		if (event.getGroup().equals("twitch") && (event.getKey().equals("channel") || event.getKey().equals("oauth") || event.getKey().equals("nick")))
+		{
+			twitchShutdown(irc);
+			irc = twitchSetup(config.oauthToken(), config.channelName(), config.nickName());
+		}
+	}
+
+	public boolean isChatboxTransparent()
+	{
+		return client.isResized() && client.getVar(Varbits.TRANSPARENT_CHATBOX) == 1;
+	}
+
+	public Client twitchSetup(String o, String nick, String name)
+	{
+
+		Client ircClient = Client.builder()
+				.serverHost("irc.chat.twitch.tv").serverPort(443)
+				.serverPassword(o)
+				.nick(nick)
+				.messageSendingQueueSupplier(TwitchDelaySender.getSupplier(false))
+				.build();
+		TwitchListener listener = new TwitchListener(ircClient);
+		ircClient.getEventManager().registerEventListener(listener);
+		ircClient.getEventManager().registerEventListener(this);
+		ircClient.connect();
+		ircClient.addChannel("#" + name);
+		return ircClient;
+	}
+
+	//Send messages to twitch chat using command "t"
+	@Subscribe
+	public void onCommand(CommandExecuted m)
+	{
+		String[] args = m.getArguments();
+		String message = "";
```

Comment:
Change message type to a StringBuilder since you are updating the variable in a loop